request-cert/main.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Marc berhault (marc@cockroachlabs.com)
+
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	certificateType = flag.String("type", "", "certificate type: node or client")
+
+	addresses = flag.String("addresses", "", "comma-separated list of DNS names and IP addresses for node certificate")
+	user      = flag.String("user", "", "username for client certificate")
+
+	certsDir        = flag.String("certs-dir", "cockroach-certs", "certs directory")
+	keySize         = flag.Int("key-size", 2048, "RSA key size in bits")
+	symlinkCASource = flag.String("symlink-ca-from", "", "if non-empty, create <certs-dir>/ca.crt linking to this file")
+)
+
+func main() {
+	flag.Parse()
+	flag.Lookup("logtostderr").Value.Set("true")
```

Comment:
yup. k8s uses glog. I was running into issues on a minimalist docker image with writes to a non-existent `/tmp/`.